src/invidious/yt_backend/extractors.cr

```diff
@@ -550,6 +551,39 @@ private module Parsers
       return {{@type.name}}
     end
   end
+
+  # Parses an InnerTube hashtagHeaderRender into a HashtagHeaderRender.
+  # Returns nil when the given object isn't a hashtagHeaderRender.
+  #
+  # hashtagHeaderRender contains metadate of the hashtag page such as video count and channel count
+  #
+  module HashtagHeaderRenderer
+    def self.process(item : JSON::Any, author_fallback : AuthorFallback)
+      if item_contents = item["hashtagHeaderRenderer"]?
+        return self.parse(item_contents)
+      end
+    end
+
+    private def self.parse(item_contents)
+      info = extract_text(item_contents.dig?("hashtagInfoText")) || ""
+
+      regex_match = /(?<videos>\d+\S)\D+(?<channels>\d+\S)/.match(info)
+
+      hashtag = extract_text(item_contents.dig?("hashtag")) || ""
+      videos = short_text_to_number(regex_match.try &.["videos"]?.try &.to_s || "0")
+      channels = short_text_to_number(regex_match.try &.["channels"]?.try &.to_s || "0")
+
+      return HashtagHeader.new({
+        tag:           hashtag,
+        channel_count: channels,
+        video_count:   videos,
+      })
+    end
+
+    def self.parser_name
+      return {{@type.name}}
+    end
+  end
```

Comment:
I'm still undecided whether making a dedicated extractor module is needed or not, as this renderer is only used for hashtags. IMO, one-off parsing makes more sense, and desn't impact the other consumers of `extract_items` (I'm worried that adding too much renderers module could impact performance long term).

Reply:
Moved the parsing to hashtag.cr